### ISSUE TITLE
Arrow render error in Text or RichText

### ISSUE THREAD

**Issue by EricMoin** [flutter/flutter#154468]
### Steps to reproduce

- In android platform,input different arrow character can lead to wrong rendering effects.
- for example:
←↑→↓↔
⬅➡
- Github can show correct effect.

### Expected results

- expected results:
←↑→↓↔
⬅➡

### Actual results

- But in Flutter,it shows:
- Fllowing is my app.
![20240831-175020]/(https://github.com/user-attachments/assets/e0398bf0-4a37-4bcf-a3e1-741834e91e66)
- Fllowing is Xianyu(Chinese PoshmarkInc)
![20240831-175327](https://github.com/user-attachments/assets/3ad419b3-f7c6-4be7-b797-c711453028f0)

### Code sample

- null

### Flutter Doctor output

[√] Flutter (Channel stable, 3.22.1, on Microsoft Windows [版本 10.0.22631.4112], locale zh-CN)
    • Flutter version 3.22.1 on channel stable at D:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a14f74ff3a (3 months ago), 2024-05-22 11:08:21 -0500
    • Engine revision 55eae6864b
    • Dart version 3.4.1
    • DevTools version 2.34.3
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\wu_mi\AppData\Local\Android\sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\\Program Files\\Android\\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11609105)
    • All Android licenses accepted.

**Comment by SuicaLondon** [comment]
For the problem `↔`, I can not reproduce on my Pixel phone and Samsung phone. Maybe you should provide more information on your device and system version?
![Screenshot_20240901-001153](https://github.com/user-attachments/assets/393c83ed-27d2-4527-8c51-e0a1d5a264e0)

But all my devices can reproduce the problem of `⬅➡`
![Screenshot 2024-09-01 at 12 21 25 AM](https://github.com/user-attachments/assets/af762460-9b44-4a57-97c1-e39f4f06bec6)


Also, it seems the problem is also happening on VScode. I observed couples of strange behaviors.
1. The size of `⬅➡` is not equal in the VScode.
2. When I paste `⬅` to other place, it may become `⬅️`.
3. If I remove the `←↑→↓↔` from `←↑→↓↔⬅➡` only leave one of `⬅➡`. It will display `⬅️➡️` on the Android devices. 

> The punny code I get from `⬅` is `xn--94i` which is the same as `⬅️`

VScode
<img width="441" alt="Screenshot 2024-09-01 at 12 26 02 AM" src="https://github.com/user-attachments/assets/283bf97d-a1f4-4f75-9622-019971816dca">

Pixel
![Screenshot 2024-09-01 at 12 28 24 AM](https://github.com/user-attachments/assets/4a95219c-214b-4c15-8004-c3819582d963)

iPhone
![Simulator Screenshot - iPhone 15 Pro Max - 2024-09-01 at 00 33 48](https://github.com/user-attachments/assets/2ec3d386-c93f-48fe-9d42-702338802e6a)

In conclusion, the problem right now is characters ⬅️ (xn--94i) and ➡️ (xn--hgi) are not showing correctly on the Android devices. It may display as `⬅➡` in various size when it came with other characters in a `Text` widget.

**Comment by EricMoin** [comment]
But the interesting thing is: if you input left arrow and right arrow in TextField, both of them will change size as soon as  you have inputed.

https://github.com/user-attachments/assets/4e6945a9-5910-46e0-a9b3-e50c52090bb7

**Comment by EricMoin** [comment]
But the interesting thing is: if you input left arrow and right arrow in TextField, both of them will change size as soon as  you have inputed.

https://github.com/user-attachments/assets/e2cde952-99eb-469b-9ef4-84837be23963

**Comment by iapicca** [comment]
> But the interesting thing is: if you input left arrow and right arrow in TextField, both of them will change size as soon as you have inputed.

@EricMoin 
given that the issue [as stated](https://github.com/flutter/flutter/issues/154468#issue-2498760655) is [not reproducible](https://github.com/flutter/flutter/issues/154468#issuecomment-2323072394) 
it would be useful to have a reproducible code sample 
for the TextField-related problem you [mentioned](https://github.com/flutter/flutter/issues/154468#issuecomment-2323208078)

**Comment by EricMoin** [comment]
While,the code is:
``` dart
Container(
        padding: EdgeInsets.symmetric(horizontal: 15.w),
        child: TextField(
          maxLines: null,
          controller: _bodyController,
          style: Get.theme.textTheme.bodyMedium,
          decoration: InputDecoration(
            hintText: LanguageMessage.editArticleContentHint,
            border: InputBorder.none,
            filled: false,
          ),
        ),
      );
```
- And in my theme.dart:
 inputDecorationTheme: InputDecorationTheme(
            contentPadding: const EdgeInsets.only(top: 0, bottom: 0),
            border: OutlineInputBorder(
                borderRadius: BorderRadius.circular(10.r),
                borderSide: const BorderSide(color: Colors.transparent)),
            enabledBorder: OutlineInputBorder(
              borderRadius: BorderRadius.circular(10.r),
              borderSide: const BorderSide(color: Colors.transparent),
            ),
            disabledBorder: InputBorder.none,
            focusedBorder: OutlineInputBorder(
                borderRadius: BorderRadius.circular(10.r),
                borderSide: const BorderSide(color: Colors.transparent)),
            filled: true,
            fillColor: const Color.fromRGBO(242, 242, 242, 1),
            hintStyle: const TextStyle(
              color: Colors.grey,
            )),

**Comment by danagbemava-nc** [comment]
Hi @EricMoin, please provide a complete minimal reproducible code sample without any third party packages so that we can properly investigate this issue. 

Thank you

**Comment by EricMoin** [comment]
Well,It's easy to reproduce it.

https://github.com/user-attachments/assets/8a87c35f-e086-4087-bfe5-89d4686a3d73

**Comment by danagbemava-nc** [comment]
Hi @EricMoin, are you using impeller or the skia rendering engine?

**Comment by EricMoin** [comment]
I don't know.It's just the default config.

**Comment by danagbemava-nc** [comment]
Hi @EricMoin, what keyboard are you using? On my Pixel 7 and an android emulator I just tested, those emojis are not available and the textfield renders the available arrow emojis as they are. See the recording below 


https://github.com/user-attachments/assets/7fc6ab02-4263-471b-92c8-50be6f4d0f4d

**Comment by EricMoin** [comment]
I use "new hardware profile" to create a new device.I think that it's not a problem which is about the phone.Because many users 
 with different devices can reproduce it easily.When I change Roboto Font to PingFang-SC or other fonts,the arrows  can shows correctly.  Maybe Roboto leads to the problem.

**Comment by danagbemava-nc** [comment]
Hi @EricMoin, without a complete sample we can't make any progress on this issue. A colleague just tested and reported seeing the same as I did in https://github.com/flutter/flutter/issues/154468#issuecomment-2328048234. 

Please provide a complete minimal reproducible code sample so that we can properly address this issue. 

Thank you

**Comment by EricMoin** [comment]
- Text('''←↑→↓↔⬅➡''')
- It meets wrong in Xianyu.
![Screenshot_2024-09-05-16-03-37-090_com taobao idl](https://github.com/user-attachments/assets/b66d81ba-f90e-4a4a-bfd4-123d2d344ffe)
- My physical device is: BlackShark 5 Pro.
- But when i try the text you give in  Pixel 3 or Pixel 7,it's no problem.
- Have you tried to create a new virsual device by "new hardware profile"?
-  I can reproduce the arrow problem in Xiaomi 14 and other Vivo phones.
- You can see the example of @SuicaLondon ( above ) ,He/She can reproduce the problem too.
- My test code is just:
- main.dart
``` dart
import 'package:flutter/material.dart';
import 'package:untitled1/home_page.dart';

void main(){
  runApp(MyApp());
}
class MyApp extends StatelessWidget{
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: HomePage(),
    );
  }
}
```
 - home_page.dart
``` dart

import 'package:flutter/material.dart';

class HomePage extends StatefulWidget{
  @override
  State<StatefulWidget> createState() => HomePageState();
}
class HomePageState extends State<HomePage>{
  @override
  void initState() {
    super.initState();
  }
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: TextField(
          maxLines: null,
        ),
      )
    );
  }
}
```

**Comment by SuicaLondon** [comment]
@danagbemava-nc @danagbemava-nc I apologize for the delay in my response. I’ve confirmed that the issue can be easily reproduced on Android, particularly on my two Pixel phones. I haven’t installed any third-party keyboards, and one of my Pixel phones is only a few days old. The problem is happening on hardcoded `Text` and `RichText`.

Here is my quick sample of code 

```Dart
class App extends StatefulWidget {
  const App({super.key});

  @override
  State<App> createState() => _AppState();
}

class _AppState extends State<App> {
  final ValueNotifier<String> _text = ValueNotifier('');
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: SafeArea(
        child: Scaffold(
          body: Column(
            children: [
              const Text('←↑→↓↔'),
              const Text('←↑→↓↔⬅➡'),
              const Text('←↑→↓↔⬅️➡️'),
              const Text('⬅➡'),
              const Text('⬅'),
              const Text('➡'),
              TextField(
                onChanged: (value) => _text.value = value,
              ),
              ValueListenableBuilder(
                valueListenable: _text,
                builder: (context, text, child) {
                  return Text(text);
                },
              )
            ],
          ),
        ),
      ),
    );
  }
}
```

<img width="452" alt="Screenshot 2024-09-05 at 8 02 17 PM" src="https://github.com/user-attachments/assets/4ef9c49f-7e56-4b34-b781-a478a0871c66">

![Screenshot_1725562844](https://github.com/user-attachments/assets/3f71f75a-dc57-4127-81fe-f775cafcaf5c)

Please compare all `⬅`, `➡`, `➡️`, `⬅️`. 
<img width="300" alt="Screenshot 2024-09-05 at 8 03 22 PM" src="https://github.com/user-attachments/assets/803d80c8-516c-411a-9229-09c547f92067">

This problem may be caused by the reflection between punny code and symbol. Please check the detail on my [comment](https://github.com/flutter/flutter/issues/154468#issuecomment-2323072394). 

<img width="757" alt="Screenshot 2024-09-05 at 8 09 51 PM" src="https://github.com/user-attachments/assets/df0b9398-184d-4999-85c4-a81426746235">

**Comment by huycozy** [comment]
Using @SuicaLondon sample code above, I can see the same result on my Pixel 7, Android 14. 

The `left` and `right` arrows don't seem to render correctly on Flutter Android app (comparing to Flutter iOS app). But I see the issue appears on Android native with TextView as well. Please see screenshots.

Keeping this to consult insight from others. Seeing this on Flutter stable 3.24.2 and master 3.25.0-1.0.pre.283.

<details open>
<summary>Demo screenshot/video</summary>

| Flutter Android ❗ | Flutter iOS ✅ |
| --------------- | --------------- |
<img width="400" src="https://github.com/user-attachments/assets/1cbf6cc0-730c-42e8-a48f-cd628cb914af"> | <img width="400" src="https://github.com/user-attachments/assets/92f8bd51-29fa-4a1b-9572-ad313279cc74"> 

| Android native ❗ | 
| --------------- |
<img width="400" src="https://github.com/user-attachments/assets/b0339b98-09ee-412b-b324-c5edea103d5d">



✅: No Issue ❗: Issue reproduced

</details>

**Comment by jonahwilliams** [comment]
If you need consistent emojis across all devices, I'd recommend shipping an emoji font with your app. I don't Flutter will ever bundle its own as that is contrary to our goal of reducing engine size.
